src/page/MainPage.tsx

```diff
@@ -1,32 +1,48 @@
 import React, { FunctionComponent, useState } from "react";
-import { CancelBlock } from "../components/CancelBlock";
 import { Header } from "../components/Header";
 import { IssueBlock } from "../components/IssueBlock";
 import { RevokeBlock } from "../components/RevokeBlock";
+import { CancelBlock } from "../components/CancelBlock";
 import { StoreDeployBlock } from "../components/StoreDeployBlock";
 
-const blocks = [
-  {
-    trigger: "issue",
-    text: "Issue Certificates",
-  },
-  {
-    trigger: "revoke",
-    text: "Revoke Certificate",
-  },
-  {
-    trigger: "cancel",
-    text: "Cancel Pending Transaction",
-  },
-];
+interface IssueType {
+  trigger: "issue";
+  text: "Issue Certificates";
+}
+
+interface RevokeType {
+  trigger: "revoke";
+  text: "Revoke Certificate";
+}
+
+interface CancelType {
+  trigger: "cancel";
+  text: "Cancel Pending Transaction";
+}
+
+type blocksType = IssueType | RevokeType | CancelType;
```

Comment:
Let's use the union on he trigger only, the text doesn't matter as we will not use it as index

```ts
type BlockType = {
  trigger: "issue" | "revoke" | "cancel"
  text:string
}
```

Reply:
done.